src/ser/helpers.rs

```diff
@@ -0,0 +1,354 @@
+        value.serialize(self)
+    }
+
+    fn serialize_unit(self) -> Result<Self::Ok, Self::Error> {
+        Ok(())
+    }
+
+    fn serialize_unit_struct(self, name: &'static str) -> Result<Self::Ok, Self::Error> {
+        Err(DummyError)
+    }
+
+    fn serialize_unit_variant(
+        self,
+        name: &'static str,
+        variant_index: u32,
+        variant: &'static str,
+    ) -> Result<Self::Ok, Self::Error> {
+        Err(DummyError)
+    }
+
+    fn serialize_newtype_struct<T: ?Sized + Serialize>(
+        self,
+        name: &'static str,
+        value: &T,
+    ) -> Result<Self::Ok, Self::Error> {
+        Err(DummyError)
+    }
+
+    fn serialize_newtype_variant<T: ?Sized + Serialize>(
+        self,
+        name: &'static str,
+        variant_index: u32,
+        variant: &'static str,
+        value: &T,
+    ) -> Result<Self::Ok, Self::Error> {
+        Err(DummyError)
+    }
+
+    fn serialize_seq(self, len: Option<usize>) -> Result<Self::SerializeSeq, Self::Error> {
+        // FIXME: Is a sequence of sequences okay here?
+        Ok(DummySer)
+    }
+
+    fn serialize_tuple(self, len: usize) -> Result<Self::SerializeTuple, Self::Error> {
+        // FIXME: I think a tuple counts as a "primitive"... doesn't it?
+        Ok(DummySer)
+    }
+
+    fn serialize_tuple_struct(
+        self,
+        name: &'static str,
+        len: usize,
+    ) -> Result<Self::SerializeTupleStruct, Self::Error> {
+        Err(DummyError)
+    }
+
+    fn serialize_tuple_variant(
+        self,
+        name: &'static str,
+        variant_index: u32,
+        variant: &'static str,
+        len: usize,
+    ) -> Result<Self::SerializeTupleVariant, Self::Error> {
+        Err(DummyError)
+    }
+
+    fn serialize_map(self, len: Option<usize>) -> Result<Self::SerializeMap, Self::Error> {
+        Err(DummyError)
+    }
+
+    fn serialize_struct(self, name: &'static str, len: usize) -> Result<Self::SerializeStruct, Self::Error> {
+        Err(DummyError)
+    }
+
+    fn serialize_struct_variant(
+        self,
+        name: &'static str,
+        variant_index: u32,
+        variant: &'static str,
+        len: usize,
+    ) -> Result<Self::SerializeStructVariant, Self::Error> {
+        Err(DummyError)
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+struct DummyError;
+
+impl StdError for DummyError {
+    fn description(&self) -> &'static str {
+        "dummy error"
+    }
+}
+
+impl Display for DummyError {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "ignore me")
+    }
+}
+
+impl SerError for DummyError {
+    fn custom<D: Display>(_: D) -> Self {
+        DummyError
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+struct DummySer;
+
+impl SerializeSeq for DummySer {
+    type Ok = ();
+    type Error = DummyError;
+
+    fn serialize_element<T>(&mut self, elem: &T) -> Result<Self::Ok, Self::Error> 
+    where T: Serialize + ?Sized {
+        if is_wrapped(elem) {
```

Comment:
Say I were to serialize `vec![1, 2, 3, 4]`, you'd end up calling `serialize_element()` on each of the elements. Usually this would be a no-op `elem.serialize(&mut *self.parent)` to serialize the element, however if you do this on a sequence of "primitives" (i.e. anything which isn't "wrapped" in some sort of named tag), you end up getting `1234` out.

I *could* wrap any primitives in an arbitrarily named tag and get something like `<value>1</value>...<value>4</value>`, but then that "value" tag is completely arbitrary, not really changeable, and probably wouldn't interop well with other XML libraries from other languages.

My `is_wrapped()` check there is aiming to prevent this issue by saying sequences of bare primitives aren't valid XML, so if you want to serialize them you need to explicitly wrap it in a named newtype.